docs/articles/retrieval_augmented_generation_eval_qdrant_arize.md

```diff
@@ -0,0 +1,519 @@
+
+![../assets/use_cases/retrieval_augmented_generation_eval_qdrant_arize/naive_rag_run_trace_detail_eval.png](../assets/use_cases/retrieval_augmented_generation_eval_qdrant_arize/naive_rag_run_trace_detail_eval.png)
+
+This can happen for various reasons, one of which could be **domain / use case specific terminology** where we may **require exact matches**.
+
+So, our next attempt will be to enrich our retriever to be powered by **Dense** as well as **Sparse vectors** to **retrieve content based on semantic similarity as well as the exact matches**.
+
+**Building a Hybrid RAG**
+
+Lets build a new vector index for the same.
+
+```python
+## Define a new collection to store your hybrid emebeddings
+COLLECTION_NAME_HYBRID = "qdrant_docs_arize_hybrid"
+```
+
+For this purpose , we’ll need to use a sparse vector model , we’ll use [Splade++](https://huggingface.co/prithivida/Splade_PP_en_v1) in this case supported through Fastembed as we can verify below :
+
+```python
+##List of supported sparse vector models
+from fastembed.sparse.sparse_text_embedding import SparseTextEmbedding
+SparseTextEmbedding.list_supported_models()
+
+Outputs
+{'model': 'prithivida/Splade_PP_en_v1',
+  'vocab_size': 30522,
+  'description': 'Independent Implementation of SPLADE++ Model for English',
+  'size_in_GB': 0.532,
+  'sources': {'hf': 'Qdrant/SPLADE_PP_en_v1'}}
+```
+
+We ingest sparse and dense vectors into Qdrant Collection.
+
+We are using **Splade++ model for Sparse Vector** Model and default Fastembed model - **bge-small-en-1.5 for Dense embeddings**.
+
+```python
+## Initializing the space to work with llama-index and related settings
+import llama_index
+from llama_index.core import Settings
+from llama_index.vector_stores.qdrant import QdrantVectorStore
+from fastembed.sparse.sparse_text_embedding import SparseTextEmbedding, SparseEmbedding
+from llama_index.embeddings.fastembed import FastEmbedEmbedding
+from typing import List, Tuple
+
+sparse_model_name = "prithivida/Splade_PP_en_v1"
+
+# This triggers the model download
+sparse_model = SparseTextEmbedding(model_name=sparse_model_name, batch_size=32)
+
+## Computing sparse vectors
+def compute_sparse_vectors(
+    texts: List[str],
+    ) -> Tuple[List[List[int]], List[List[float]]]:
+    indices, values = [], []
+    for embedding in sparse_model.embed(texts):
+        indices.append(embedding.indices.tolist())
+        values.append(embedding.values.tolist())
+    return indices, values
+
+## Creating a vector store with Hybrid search enabled
+hybrid_vector_store = QdrantVectorStore(
+    client=client,
+    collection_name=COLLECTION_NAME_HYBRID,
+    enable_hybrid=True,
+    sparse_doc_fn=compute_sparse_vectors,
+    sparse_query_fn=compute_sparse_vectors)
+
+storage_context = StorageContext.from_defaults(vector_store=hybrid_vector_store)
+
+Settings.embed_model = FastEmbedEmbedding(model_name="BAAI/bge-small-en-v1.5")
+```
```

Comment:
Shouldn't above snippet include dense vector computation?

Reply:
We have done that before in the first experiment , this is where we want to showcase what sparse vector bring to the equation and thus we show sparse vector in isolation, followed by Hybrid approach below.

Reply:
@atarora
Shouldn't above snippet include dense vector computation?